scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -2053,6 +2053,49 @@ def scc_order(self) -> List[List[Node]]:
         except Exception as e:
             raise EDTError(e)
 
+    def _link_properties(self, parent_node, node):
+        """
+        Link node properties to parent_node. Iterates over child nodes
+        recursively if necessary.
```

Comment:
The purpose of the function is not clear from its name and docstring; can you please expand this? The arguments are `parent_node` and `node` but the caller is passing `node, node`. This feels confusingly written and the scope is a bit muddy.

Reply:
Fair enough, I've renamed functions and variables, added few commments, reworded the function comment and added a wrapper for the first call. Let me know if that's clear enough.